### PR TITLE
Refactor `profile tcprtt` unit test

### PR DESCRIPTION
**Pull request by eiffel-fl**
Hi.

This PR is a followup to this [comment](https://github.com/inspektor-gadget/inspektor-gadget/pull/1717#discussion_r1229261601) and refactors the `profile tcprtt` unit tests to avoid duplication between IPv4 and IPv6 tests.

Best regards.

### CODE REVIEW

File: pkg/gadgets/profile/tcprtt/tracer/tracer_test.go [hunk 1]
```diff
@@ -350,219 +352,7 @@ func TestRunWithResultV4(t *testing.T) {
...
-			getGadgetParams: func() *params.Params {
-				params := gadget.ParamDescs().ToParams()
-				params.Get(ParamFilterLocalAddress).Set(clientIP.String())
-				return params
-			},
-			expectedResult: expectedResult{
-				exactHistograms: 1,
-			},
-		},
-		"filter_by_invalid_laddr": {
```
Comment (mauriciovasquezbernal) [inline pkg/gadgets/profile/tcprtt/tracer/tracer_test.go:429]: I think this name is misleading as the IP is not invalid but nonexisting.

**Review by mauriciovasquezbernal — COMMENTED** [review]
What about something like https://github.com/inspektor-gadget/inspektor-gadget/commit/5d02ebdcfa3b5e3c76bcd58552a2003c0bbad38a to further reduce the duplication?

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM. It'd be nice to update the first commit message since it's outdated.

File: pkg/gadgets/profile/tcprtt/tracer/bpf/tcprtt.bpf.c [hunk 2]
```diff
@@ -114,6 +114,12 @@ static int handle_tcp_rcv_established(struct sock *sk)
 		if (key.family == AF_INET6)
 			bpf_probe_read_kernel(&key.addr, sizeof(key.addr), BPF_CORE_READ(inet, pinet6, saddr.in6_u.u6_addr8));
 		else
+			/*
+			 * It is fine to use "->" operator with bpf_probe_read_kernel() as we are
+			 * using vmlinux.h which defines struct with preserve_access_index
+			 * attribute, see:
+			 * https://nakryiko.com/posts/bpf-core-reference-guide/#defining-own-co-re-relocatable-type-definitions
+			 */
```
Comment (blanquicet) [inline pkg/gadgets/profile/tcprtt/tracer/bpf/tcprtt.bpf.c:122]: Thanks for this comment.

File: pkg/gadgets/profile/tcprtt/tracer/tracer_test.go [hunk 3]
```diff
@@ -326,22 +326,24 @@ func TestParseParams(t *testing.T) {
...
-		serverPort = 8080
-	)
+type testDefinition struct {
+	getGadgetParams func() *params.Params
+	timeout         time.Duration
+	expectedResult  expectedResult
+}
 
+func testRunWithResult(t *testing.T, serverPort int, serverIP, clientIP net.IP, ipVersion int) {
 	// TODO: Use random IPs.
```
Comment (blanquicet) [inline pkg/gadgets/profile/tcprtt/tracer/tracer_test.go:345]: nit: This comment should go in `TestRunWithResultV{4|6}`.

**Review by blanquicet — APPROVED** [review]
LGTM!

**Pull request merged**